src/components/ui/Tag/index.js

```diff
@@ -0,0 +1,147 @@
+  const { children, label, onDismiss, ...otherProps } = props;
+
+  return (
+    <span ref={ref} {...otherProps}>
+      {label && (
+        <span // TODO: switch with VisuallyHidden component, once available.
+          css={css`
+            font-style: bold;
+            position: absolute;
+            left: -10000px;
+            top: auto;
+            width: 1px;
+            height: 1px;
+            overflow: hidden;
+          `}
+        >
+          {label}:
+        </span>
+      )}
+
+      {children}
+      {onDismiss && (
+        <button
+          type="button"
+          onClick={onDismiss}
+          css={css`
+            box-shadow: 0;
+            border: 0;
+            background: none;
+            color: currentColor;
+            cursor: pointer;
+            vertical-align: 0.2rem;
+            margin: 0;
+            padding: 0;
+
+            &:hover {
+              opacity: 0.75;
+            }
+          `}
+        >
+          <Icon
+            name="x"
+            small
+            css={css`
+              display: inline-block;
+              margin-left: ${toUnits(theme.spacing.padding.small)};
+              margin-top: ${toUnits(theme.spacing.padding.xxSmall)};
+              opacity: 1;
+              vertical-align: middle;
+            `}
+          />
+        </button>
+      )}
+    </span>
+  );
+});
+
+export const styles = (props) => {
+  const { color, count, status, theme } = props;
+  let textColor = theme.colors.text.light;
+  let backgroundColor;
+
+  if (color) {
+    backgroundColor = color;
+  }
+
+  if (status) {
+    backgroundColor = theme.colors.intent[status];
+  }
+
+  if (status || color) {
+    textColor = getContrastingTextColor({ color: backgroundColor, theme });
+  }
+
+  return css`
+    ${metadata.small(theme)};
+    color: ${textColor};
+    margin: 0 ${toUnits(theme.spacing.padding.xSmall)}
+      ${toUnits(theme.spacing.padding.xSmall)} 0;
+
+    ${backgroundColor &&
+      css`
+        background: ${backgroundColor};
+        padding: ${toUnits(theme.spacing.padding.xSmall)}
+          ${toUnits(theme.spacing.padding.small)};
+      `}
+
+    ${count &&
+      css`
+        border-radius: ${toUnits(theme.spacing.padding.xLarge)};
+        min-width: ${toUnits(theme.spacing.padding.xLarge)};
+        height: ${toUnits(theme.spacing.padding.xLarge)};
+        display: inline-flex;
+        align-items: center;
+        justify-content: center;
+        box-sizing: border-box;
+      `}
+  `;
+};
+
+Tag.defaultProps = {
+  children: undefined,
+  color: null,
+  label: null,
+  count: false,
+  onDismiss: null,
+  status: undefined,
+};
+
+Tag.propTypes = {
+  /** @ignore */
+  children: PropTypes.node,
+  /** Use colour to differentiate different tags or indicate status. */
+  color: PropTypes.string,
+  /** Optional label for providing additional context to screen reader users. */
+  label: PropTypes.string,
+  /** Style numbers in a more badge-y kind of way. */
```

Comment:
We can tweak this explanation I think—it's a very cute Sarah-ism, but might read strangely to an ESL user 😄